daemon/algod/api/server/v2/handlers.go

```diff
@@ -1132,14 +1133,27 @@ func (v2 *Handlers) AbortCatchup(ctx echo.Context, catchpoint string) error {
 
 // TealCompile compiles TEAL code to binary, return both binary and hash
 // (POST /v2/teal/compile)
-func (v2 *Handlers) TealCompile(ctx echo.Context) error {
-	// return early if teal compile is not allowed in node config
+func (v2 *Handlers) TealCompile(ctx echo.Context, params generated.TealCompileParams) (err error) {
+	// Return early if teal compile is not allowed in node config.
 	if !v2.Node.Config().EnableDeveloperAPI {
 		return ctx.String(http.StatusNotFound, "/teal/compile was not enabled in the configuration file by setting the EnableDeveloperAPI to true")
 	}
+	// Check if we should return the source map.
+	sourcemapFlag := false
+	var sourcemap logic.SourceMap
```

Comment:
nitty but i prefer grouping multiple var declarations ```go
var (
 sourcemapFlag bool
 sourcemap logic.SourcceMap
)
```

Reply:
Refactored the whole section to remove some ugly code